### PR TITLE
feat: update kiota package to 24.0.4

### PR DESCRIPTION
**Pull request by Persistent13**
I have a need for the delete user API endpoint and that doesn't appear to be in the 24.0.1 OpenAPI definition but is present in the 24.0.4 version.

This change is a lot and I'm not sure if this is expected so just in case these are the steps I've taken to create this PR:

- Ran `kiota info -l CSharp`

Which output

```shell
The language CSharp is currently in Stable maturity level.
After generating code for this language, you need to install the following packages:
Package Name                             Version
Microsoft.Kiota.Abstractions             1.8.4
Microsoft.Kiota.Authentication.Azure     1.1.5
Microsoft.Kiota.Http.HttpClientLibrary   1.4.0
Microsoft.Kiota.Serialization.Form       1.1.6
Microsoft.Kiota.Serialization.Json       1.2.3
Microsoft.Kiota.Serialization.Multipart  1.1.4
Microsoft.Kiota.Serialization.Text       1.1.5

Hint: use the install command to install the dependencies.
Example:
   dotnet add package Microsoft.Kiota.Abstractions --version 1.8.4
   dotnet add package Microsoft.Kiota.Authentication.Azure --version 1.1.5
   dotnet add package Microsoft.Kiota.Http.HttpClientLibrary --version 1.4.0
   dotnet add package Microsoft.Kiota.Serialization.Form --version 1.1.6
   dotnet add package Microsoft.Kiota.Serialization.Json --version 1.2.3
   dotnet add package Microsoft.Kiota.Serialization.Multipart --version 1.1.4
   dotnet add package Microsoft.Kiota.Serialization.Text --version 1.1.5
```

I then ran the dotnet add package command above in turn

- Generated the changes with

```shell
kiota generate -l CSharp --log-level trace --output . --namespace-name Keycloak.AuthServices.Sdk.Kiota.Admin --class-name KeycloakAdminApiClient --exclude-backward-compatible --openapi https://www.keycloak.org/docs-api/24.0.1/rest-api/openapi.json
```

And now we're here! I committed the changes as is and opened the PR.

Let me know if I did anything wrong or need anything changed.

### CODE REVIEW

**Comment by NikiforovAll** [comment]
LGTM, could you please install https://csharpier.com/ and run it locally for newly generated code? 

My idea is to eliminate this kind of change in the future by using an opinionated formater. At least I want to see how it works in practice

**Comment by NikiforovAll** [comment]
Could you please bump the package version to 24.0.4 in the csproj file?

**Comment by NikiforovAll** [comment]
Did you run generation for 24.0.4 version? In the example above you mention 24.0.1 for generation

**Comment by Persistent13** [comment]
I thought I had, let me try again. I'll also get that CSharpier and version bump done as well.

**Comment by Persistent13** [comment]
I had indeed slipped in 24.0.1 in by accident. I just reran it with 24.0.4 along with the CSharpier and version bump as well.

Let me know if there's anything else!

**Comment by NikiforovAll** [comment]
Thank you for the PR
